### PR TITLE
Adding jq check of the exercise status to test runner

### LINKED ISSUE #2666: CI test job should exclude deprecated exercises
The GitHub Actions workflow that uses the java-test-runner to test all exercises in this repository currently tests _all_ exercise directories, including deprecated exercises. Since [deprecated exercises should remain untouched](https://exercism.org/docs/building/tracks/deprecated-exercises) and the java-test-runner recently dropped support for JUnit 4, the CI build is now failing on these exercises.

To resolve this, the [bin/test-with-test-runner](https://github.com/exercism/java/blob/main/bin/test-with-test-runner) script should be updated to skip deprecated exercises. Deprecated exercises are marked with `"status": "deprecated"` in the track's `config.json`, so it should be fairly simple to determine which exercises are deprecated by using `jq`.

### PR DESCRIPTION
**Pull request by manumafe98**
# pull request

Fixes #2666

The goal is to make test runner exclude deprecated exercises

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/main/POLICIES.md#event-checklist)

### CODE REVIEW

File: bin/test-with-test-runner [hunk 1]
```diff
@@ -34,6 +34,12 @@ function verify_exercise() {
     local implementation_files=($(jq -r --arg d "${dir}" --arg k "${implementation_file_key}" '$d + "/" + .files[$k][]' "${dir}/.meta/config.json"))
     local stub_files=($(jq -r --arg d "${dir}" '$d + "/" + .files.solution[]' "${dir}/.meta/config.json"))
     local results_file="${output_dir}/results.json"
+    local exercise_type=$3
+    local exercise_status=($(jq -r --arg t ${exercise_type} --arg s ${slug} '.exercises[$t][] | select(.slug == $s).status' config.json))
+
+    if [[ $exercise_status == "deprecated" ]]; then
+        return
```
Comment (sanderploegsma) [inline bin/test-with-test-runner:41]: Maybe print a message here indicating that we are skipping it. That saves future headaches when someone notices in the logs that not all exercises are being tested and needs to dive in to figure out why.

```suggestion
        echo "Skipping deprecated exercise: ${slug}"
        return
```

**Review by sanderploegsma — APPROVED** [review]
Nice work!

File: bin/test-with-test-runner (hunk 1 above)
Comment (manumafe98) [reply bin/test-with-test-runner:42]: Sure! done

**Comment by sanderploegsma** [comment]
@manumafe98 LGTM! You can merge it yourself now if you want 🚀

**Pull request merged**